parser/types.go

```diff
@@ -119,29 +119,30 @@ func incompletePipeExpr(call interface{}, text []byte, pos position) (*ast.PipeE
 func memberexprs(head, tail interface{}, text []byte, pos position) (ast.Expression, error) {
 	res := head.(ast.Expression)
 	for _, prop := range toIfaceSlice(tail) {
-		res = &ast.MemberExpression{
-			Object:   res,
-			Property: prop.(ast.Expression),
-			BaseNode: base(text, pos),
+		switch expr := prop.(type) {
+		case *ast.MemberExpression:
+			expr.BaseNode = base(text, pos)
+			expr.Object = res
+			res = expr
+		case *ast.IndexExpression:
+			expr.BaseNode = base(text, pos)
+			expr.Array = res
+			res = expr
 		}
 	}
 	return res, nil
 }
 
-func memberexpr(object, property interface{}, text []byte, pos position) (*ast.MemberExpression, error) {
-	m := &ast.MemberExpression{
-		BaseNode: base(text, pos),
-	}
-
-	if object != nil {
-		m.Object = object.(ast.Expression)
-	}
+func memberExpr(p interface{}) (*ast.MemberExpression, error) {
+	return &ast.MemberExpression{Property: p.(*ast.Identifier)}, nil
+}
 
-	if property != nil {
-		m.Property = property.(*ast.Identifier)
-	}
+func memberExprFromIndexExpr(p interface{}) (*ast.MemberExpression, error) {
```

Comment:
Can you rename this to something like `memberExprFromStringLiteral` and rename `memberExpr` to something like `memberExprFromIdentifier`?